examples/python/Boxes_QPE_example.py

```diff
@@ -0,0 +1,561 @@
+iizz = QubitPauliString(q, [Pauli.I, Pauli.I, Pauli.Z, Pauli.Z])
+
+
+# In[8]:
+
+
+qpo_dict = {QubitPauliString(): -0.0596205827603462, iiiz: 0.175759429183197, 
+            iizi: -0.236671176780357, izii: 0.175759429183197,
+            iiiz: -0.236671176780357, zizi: 0.122227149362618, iziz: 0.122227149362618,
+            zzii: 0.170015464396032, izzi: 0.167144319253372,
+            yxxy: 0.0449171698907539, xxyy: -0.0449171698907539, 
+            yyxx: -0.0449171698907539, xyyx: 0.0449171698907539,
+            ziiz: 0.167144319253372, iizz: 0.175703383319070}
+
+
+# In[9]:
+
+
+qpo_h2 = QubitPauliOperator(qpo_dict)
+
+
+# In[10]:
+
+
+from pytket.utils import gen_term_sequence_circuit
+from pytket.passes import DecomposeBoxes
+
+h2_circ = gen_term_sequence_circuit(qpo_h2, Circuit(4))
+h2_circ.name = "U"
+
+render_circuit_jupyter(h2_circ)
+
+
+# We can create a controlled unitary $U$ with a `QControlBox` with $n$ controls
+
+# In[11]:
+
+
+from pytket.circuit import QControlBox
+
+u_box = CircBox(h2_circ)
+
+
+# In[12]:
+
+
+controlled_u = QControlBox(u_box, n=1)
+
+
+# In[13]:
+
+
+test_circ = Circuit(5).add_gate(controlled_u, [0, 1, 2, 3, 4])
+render_circuit_jupyter(test_circ)
+
+
+# ## Putting it all together
+
+# In[14]:
+
+
+def build_phase_est_circuit(n_measurement_qubits: int,
+                            state_prep_circuit: Circuit,
+                            unitary_circuit: Circuit) -> Circuit:
+    qpe_circ = Circuit()
+    n_ancillas = state_prep_circuit.n_qubits
+    measurement_register = qpe_circ.add_q_register("m", n_measurement_qubits)
+    state_prep_register = qpe_circ.add_q_register("p", n_ancillas)
+    c_reg = qpe_circ.add_c_register("c", n_measurement_qubits)
+    
+    state_prep_qubits = list(state_prep_register)
+    measurement_qubits = list(measurement_register)
+    
+    qpe_circ.add_circuit(state_prep_circuit, state_prep_qubits)
+    
+    unitary_circuit.name = "U"
+    u_box = CircBox(unitary_circuit)
+    controlled_u = QControlBox(u_box, 1)
+    
+    for m_qubit in measurement_qubits:
+        qpe_circ.H(m_qubit)
+        
+    for m_qubit in range(n_measurement_qubits):
+        control_index = n_measurement_qubits - m_qubit - 1
+        control_qubit = [measurement_register[control_index]]
+        for _ in range(2**m_qubit):
+                qpe_circ.add_qcontrolbox(
+                    controlled_u, control_qubit + state_prep_qubits
+                )
+        
+    qft_box = CircBox(build_qft_circuit(n_measurement_qubits))
+    inverse_qft_box = qft_box.dagger
+    
+    qpe_circ.add_circbox(inverse_qft_box, measurement_qubits)
+    
+    qpe_circ.measure_register(measurement_register, "c")
+    
+    return qpe_circ
+
+
+# ## Phase Estimation with a Trivial Eigenstate
+# 
+# Lets test our circuit construction by preparing a trivial $|1\rangle $eigenstate of the $\text{U1}$ gate. We can then see if our phase estimation circuit returns the expected eigenvalue.
```

Comment:
This equation isn't compiling in the notebook for me, is it fine for you?

Reply:
Hmm, seems to look okay for me. Will rerun to be sure